### PR TITLE
Improve golang module dependencies

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
These changes will make it easier to import CDI from Kubevirt. 

Changes:
- `prometheus-operator` - depend only on submodule `/pkg/apis/monitoring` instead of the whole project
- OLM - depend on the API repo `github.com/operator-framework/api` instead of the OLM repo
- `k8s.io/*` - updated `client-go` to `v0.23.5` instead of `v12.0.0+incompatible`, which is inconvenient when CDI is imported by Kubevirt.
- `controller-lifecycle-operator-sdk` - use tag for the `/api` submodule

**Release note**:
```release-note
None
```

### CODE REVIEW

File: cmd/cdi-operator/BUILD.bazel [hunk 1]
```diff
@@ -9,10 +9,10 @@ go_library(
         "//pkg/operator/controller:go_default_library",
         "//pkg/util:go_default_library",
         "//staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1:go_default_library",
-        "//vendor/github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1:go_default_library",
         "//vendor/github.com/openshift/api/config/v1:go_default_library",
         "//vendor/github.com/openshift/api/route/v1:go_default_library",
         "//vendor/github.com/openshift/api/security/v1:go_default_library",
+        "//vendor/github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1:go_default_library",
```
Comment (brybacki) [inline cmd/cdi-operator/BUILD.bazel:15]: what exactly is a difference between github.com/prometheus-operator and github.com/coreos ?

File: go.mod [hunk 2]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/openshift/client-go v0.0.0
 	github.com/openshift/custom-resource-status v1.1.2
 	github.com/openshift/library-go v0.0.0-20211220195323-eca2c467c492
-	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20190725173916-b56e63a643cc
+	github.com/operator-framework/api v0.11.1
```
Comment (brybacki) [inline go.mod:27]: nice!

File: pkg/operator/resources/operator/operator.go [hunk 3]
```diff
@@ -395,21 +396,6 @@ func createPrometheusPorts() []corev1.ContainerPort {
 	}
 }
 
-type csvPermissions struct {
```
Comment (brybacki) [inline pkg/operator/resources/operator/operator.go:398]: interesting that it was internal before

**Comment by brybacki** [comment]
@akrejcir thanks for the PR, This is really a good cleanup!

/lgtm

**Comment by brybacki** [comment]
lets see if it works, and the tests are green

File: go.mod [hunk 4]
```diff
@@ -148,35 +148,8 @@ replace (
...
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.23.0
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.23.0
-	k8s.io/kubectl => k8s.io/kubectl v0.23.0
-	k8s.io/kubelet => k8s.io/kubelet v0.23.0
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.23.0
-	k8s.io/metrics => k8s.io/metrics v0.23.0
-	k8s.io/mount-utils => k8s.io/mount-utils v0.23.0
-	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.23.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.23.0
```
Comment (mhenriks) [inline go.mod:179]: We want to keep these versions pinned.  CDI is not just a library project.

**Comment by mhenriks** [comment]
/hold

we must keep k8s deps pinned

File: cmd/cdi-operator/BUILD.bazel (hunk 1 above)
Comment (akrejcir) [reply cmd/cdi-operator/BUILD.bazel:15]: There is no difference. The URL "github.com/coreos/prometheus-operator" redirects to "github.com/prometheus-operator/prometheus-operator".

File: go.mod (hunk 4 above)
Comment (akrejcir) [reply go.mod]: Ok, I returned the `replace` lines.

**Comment by mhenriks** [comment]
Looks like KubeVirt only imports the [pkg/clone](https://github.com/kubevirt/kubevirt/tree/main/vendor/kubevirt.io/containerized-data-importer) package.  Maybe make that module?

**Comment by brybacki** [comment]
> Looks like KubeVirt only imports the [pkg/clone](https://github.com/kubevirt/kubevirt/tree/main/vendor/kubevirt.io/containerized-data-importer) package. Maybe make that module?

Does it change anything in dependency tree/analysis when that is a module? If yes then I agree with your comment.

**Comment by mhenriks** [comment]
> > Looks like KubeVirt only imports the [pkg/clone](https://github.com/kubevirt/kubevirt/tree/main/vendor/kubevirt.io/containerized-data-importer) package. Maybe make that module?
> 
> Does it change anything in dependency tree/analysis when that is a module? If yes then I agree with your comment.

Should, these are the only packages it references:  https://github.com/kubevirt/kubevirt/blob/main/vendor/kubevirt.io/containerized-data-importer/pkg/clone/BUILD.bazel#L10-L13

**Comment by brybacki** [comment]
@akrejcir this is very good PR, and making a [pkg/clone](https://github.com/kubevirt/kubevirt/tree/main/vendor/kubevirt.io/containerized-data-importer) a module will make it even better. That way managinf dependencies in kubevirt will be easier/cleaner.
What is your opinion on this?

**Comment by akrejcir** [comment]
I think it is up to you to decide if it's worth making a submodule. It would have it's own `vendor` directory with all the dependencies, so some files would be duplicated in this repository. And every time you make a new release, you would need to create an annotated tag `vX.Y.Z/pkg/clone`.

**Comment by brybacki** [comment]
Its own `version` and `tag`. I did not think about this.

@mhenriks @awels what do you think?

**Pull request closed without merging**